src/docs/perf/deferred-components.md

```diff
@@ -0,0 +1,591 @@
+
+<!-- skip -->
+```dart
+// box.dart
+
+import 'package:flutter/widgets.dart';
+
+/// A simple blue 30x30 box.
+class DeferredBox extends StatelessWidget {
+  DeferredBox() {}
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      height: 30,
+      width: 30,
+      color: Colors.blue,
+    );
+  }
+}
+```
+</li>
+
+<li markdown="1">Import the new Dart library
+    with the `deferred` keyword in your app and
+    call `loadLibrary()` (see [lazily loading a library][]).
+    The following example uses `FutureBuilder`
+    to wait for the `loadLibrary` `future` (created in
+    `initState`) to complete and display a
+    `CircularProgressIndicator` as a placeholder.
+    When the `future` completes, it returns the `DeferredBox` widget.
+    `SomeWidget` can then be used in the app as normal and
+    won't ever attempt to access the Dart lib until
+    it has successfully loaded.
+
+<!-- skip -->
+```dart
+import 'box.Dart' deferred as box;
+
+// ...
+
+class SomeWidget extends StatefulWidget {
+  @override
+  _SomeWidgetState createState() => _SomeWidgetState();
+}
+
+class _SomeWidgetState extends State<SomeWidget> {
+  Future<void> _libraryFuture;
+
+  @override
+  void initState() {
+    _libraryFuture = box.loadLibrary();
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return FutureBuilder<void>(
+      future: _libraryFuture,
+      builder: (BuildContext context, AsyncSnapshot<void> snapshot) {
+        if (snapshot.connectionState == ConnectionState.done) {
+          if (snapshot.hasError) {
+            return Text('Error: ${snapshot.error}');
+          }
+          return box.DeferredBox();
+        }
+        return CircularProgressIndicator();
+      },
+    );
+  }
+}
+// ...
+```
+
+The `loadLibrary()` function returns a `Future<void>`
+that completes successfully when the code in the library
+is available for use and completes with an error otherwise.
+All usage of symbols from the deferred library should be
+guarded behind a completed `loadLibrary()` call. All imports
+of the library must be marked as `deferred` for it to be
+compiled appropriately to be used in a deferred component.
+If a component has already been loaded, additional calls
+to `loadLibrary()` complete quickly (but not synchronously).
+The `loadLibrary()` function can also be called early to
+trigger a pre-load to help mask the loading time.
+
+You can find another example of deferred import loading in
+[Flutter Gallery’s lib/deferred_widget.dart][].
+</li>
+</ol>
+
+### Step 3: Building the app
+
+Use the `$ flutter build appbundle` command to
+build deferred components apps. The command
+assists you by validating that your project is
+properly set up to build deferred components apps.
+By default, the build fails if the validator detects
+any issues and guides you through suggested changes to fix them.
+
+You might opt out of building deferred components
```

Comment:
might -> may

Also, this paragraph should go in a `Note:` box as it is not part of the regular guide.